RadixWallet/Clients/EntitiesInvolvedClient/EntitiesInvolvedDataSource.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+// MARK: - EntitiesInvolvedDataSource
+protocol EntitiesInvolvedDataSource: Sendable {
+	var addressesOfPersonasRequiringAuth: [IdentityAddress] { get }
+	var addressesOfAccountsRequiringAuth: [AccountAddress] { get }
+	var addressesOfAccountsWithdrawnFrom: [AccountAddress] { get }
+	var addressesOfAccountsDepositedInto: [AccountAddress] { get }
+}
+
+// MARK: - ManifestSummary + EntitiesInvolvedDataSource
+extension ManifestSummary: EntitiesInvolvedDataSource {}
+
+// MARK: - PreAuthToReview + EntitiesInvolvedDataSource
+extension PreAuthToReview: EntitiesInvolvedDataSource {
+	var addressesOfPersonasRequiringAuth: [IdentityAddress] {
+		switch self {
+		case let .open(value):
+			value.summary.addressesOfPersonasRequiringAuth
+		case let .enclosed(value):
+			value.summary.addressesOfIdentitiesRequiringAuth
+		}
+	}
+
+	var addressesOfAccountsRequiringAuth: [AccountAddress] {
```

Comment:
This looks like a lot of effort for something we already have in Sargon? https://github.com/radixdlt/sargon/blob/main/crates/sargon/src/signing/extractor_of_entities_requiring_auth.rs#L9

Reply:
This was just code moved from `TransactionClient` ([here](https://github.com/radixdlt/babylon-wallet-ios/pull/1376/files#diff-025b8374e54eec385cf67e5e234a87934602e4fb649ec603ff243a9c35e1dffaL48-L70)), to make it reusable rather than having code duplicated. Didn't know we had the logic on Sargon on it, will look into it.

Reply:
Ok so the reason why we cannot easily use that method from Sargon is because it takes a `ManifestSummary`, and for enclosed transactions we have `ExecutionSummary`. 

We can logically extend such method to receive a `PreAuthToReview` instead, and use the corresponding `addresses_of_accounts_requiring_auth` & `addresses_of_personas_requiring_auth` from it (as I am doing here). Let me know if you think it is worth extending such code in Sargon.